api/src/main/proto/stellarstation/api/v1/groundstation/groundstation.proto

```diff
@@ -208,3 +226,71 @@ message DeleteUnavailabilityWindowRequest {
   // Currently no payload in the response.
 }
+
+// A request for the `OpenGroundStationStream` method.
+message GroundStationStreamRequest {
+  // The ID of the ground station that is opening a stream. The ID of a ground station can be found
+  // on the StellarStation Console page for the ground station.
+  string ground_station_id = 1;
+
+  // A request to send on the ground station stream. The request does not need to be filled on the
+  // first request to `OpenGroundStationStream`.
+  oneof Request {
+    SendTelemetryRequest send_telemetry_request = 2;
+
+    // An event that occurred while processing the stream. Satellite operators will often use these
+    // events for debugging and monitoring.
+    //
+    // - `command_sent` must be returned right after completing transmission of
+    //    commands received in a `SendSatelliteCommandsResponse`.
+    // - `plan_monitoring_event.ground_station_configuration` must be returned once at the beginning
+    //   of the execution of a plan and contains information about the actual configuration of the
+    //   ground station at the time.
+    // - `plan_monitoring_event.ground_station_state` must be sent continuously throughout the
+    //   plan and contains real-time monitoring of the hardware of the ground station.
+    // - `plan_monitoring_event.ground_station_event` must be sent once at each of the transitions
+    //   documented in `PlanLifecycleEvent` when executing a plan.
+    StreamEvent stream_event = 3;
+  }
+}
+
+// A request to send telemetry received from a satellite to a user.
+message SendTelemetryRequest {
+  // The telemetry being sent.
+  Telemetry telemetry = 1;
+
+  // The ID of the plan the telemetry is being sent for.
+  string plan_id = 2;
+}
+
+// A response for the `OpenGroundStationStream` method. The ground station must process a response
+// immediately when received on the stream.
+message GroundStationStreamResponse {
+  // The ID of the plan this response corresponds to. Unfilled if this response does not correspond
+  // to a plan.
+  string plan_id = 1;
+
+  // A unique ID identifying this response. If filled, any `StreamEvent` generated when handling
+  // this GroundStationStreamResponse` must have this value copied into `StreamEvent` must have this
+  // value copied into `StreamEvent.request_id`. Currently, this includes
+  //
+  // - Returning `CommandSentFromGroundStation` after completing transmission of the commands in a
+  //   `SendSatelliteCommandsResponse`.
+  string response_id = 2;
+
+  // A response sent on the stream to be processed by the ground station. Currently only commands to
+  // send to a satellite will be returned to the ground station.
+  oneof Response {
+    // A response containing commands to send to the satellite.
+    SendSatelliteCommandsResponse send_satellite_commands_response = 3;
+  }
+}
+
+// A response containing a command to transmit to the satellite.
+//
+// Next ID: 1
+message SendSatelliteCommandsResponse {
```

Comment:
It feels a bit weird to call this a response when it's kind of a request. I suppose we'd use SendSatelliteCommandsRequest directly if we wanted to keep the same name.

Do you think we are likely to have things inside GroundStationStreamResponse that are not some kind of command?

I'm mostly just musing about the name and can live with it as is.

Reply:
It's a response on the stream, though a request from the satellite operator. I don't think either is easier to understand, but response seems a bit more consistent.